.github/workflows/ci_master.yml

```
@@ -0,0 +1,162 @@
+name: nf-core CI master
+# This workflow runs the pipeline with the minimal test dataset to check that it completes without any syntax errors upon master PR
+on:
+  pull_request:
+    branches:
+      - master
+  release:
+    types: [published]
+  workflow_dispatch:
+
+env:
+  NXF_ANSI_LOG: false
+  NXF_SINGULARITY_CACHEDIR: ${{ github.workspace }}/.singularity
+  NXF_SINGULARITY_LIBRARYDIR: ${{ github.workspace }}/.singularity
+
+concurrency:
+  group: "${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}"
+  cancel-in-progress: true
+
+jobs:
+  test:
+    name: "Run pipeline with test data (${{ matrix.NXF_VER }} | ${{ matrix.test_name }} | ${{ matrix.profile }})"
+    # Only run on push if this is the nf-core dev branch (merged PRs)
+    if: "${{ github.event_name != 'push' || (github.event_name == 'push' && github.repository == 'nf-core/multiplesequencealign') }}"
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        NXF_VER:
+          - "24.10.4"
+          - "latest-everything"
+        profile:
+          - "docker"
+          - "singularity"
```

this suggestion is in case you would like to reduce code duplication, the if statement below is a bit complicated but I think it's good to reduce code duplication for maintenance in the future
```suggestion
          - "singularity"
          - "conda"
```